default_omics_by_model_and_datatype.py

```diff
@@ -0,0 +1,172 @@
+		elif filter_string_in != '' and filter_string_ex == '':
+			globals()[filteredtablename] = thistable.query(filter_string_in)
+		elif filter_string_ex != '' and filter_string_in == '':
+			globals()[filteredtablename] = thistable.query(filter_string_ex)
+		else:
+			print("should never come here")
+
+
+
+# Merge tables based on merge parameters specified in the config file
+# The merged table is stored in the variable "merged_table"
+merged_table = None
+for thistablename, thisfilteredtablename in zip(tables_to_filter_names, filtered_tables_names):
+	merge_params = configdata["table_filters"][thistablename].get("merge_parameters", None)
+	if merged_table is None:
+		merged_table = globals()[thisfilteredtablename]
+	if merge_params:
+		primary_key = merge_params["primary"][0]
+		foreign_key = merge_params["foreign"][0]
+		merged_table = pd.merge(globals()[thisfilteredtablename], 
+						  merged_table, 
+						  left_on = primary_key, 
+						  right_on = foreign_key, 
+						  how = 'inner', 
+						  suffixes=['_' + thistablename,'_merged'])
+
+
+# Add priority columns to the merged table based on the priority rankings specified in the config file
+# The priority columns are added to the merged table with the suffix "_priority"
+# The priority ladder is used for a final outcome of selecing the best row for each model, building from the bottom up.
+
+priority_rankings = configdata["priority_rankings"]
+for rankingvar, ranking in priority_rankings["rankings"].items():
+	if pd.api.types.is_bool_dtype(merged_table[rankingvar]):
+		merged_table[rankingvar].fillna(False, inplace = True)
+	elif pd.api.types.is_integer_dtype(merged_table[rankingvar]):
+		merged_table[rankingvar].fillna(1000, inplace = True)
+	elif pd.api.types.is_string_dtype(merged_table[rankingvar]):
+		merged_table[rankingvar].fillna("None", inplace = True)
+	else:
+		merged_table[rankingvar].fillna("NA", inplace = True)
+	merged_table[rankingvar + '_priority'] = merged_table[rankingvar].astype(str).map(ranking)
+
+# Special case to accommodate for the way gumbo is structured. 
+# You can have the same profile generating two sequences of the same type wgs or rna. This is something GP just delivers without asking
+# You can also have two different profiles for each CDS - this would have happened when Sam manually gave each CDS ID its own profile_id 
+# for clarity. At the moment it is unclear how many fall under each case, so filtering twice with same parameters (version)
+# Wish I could have found a way to not hardcode column names these next 3 lines, but I am not sure how to go about that.
+idx = []
+idx.append(merged_table.groupby(["sequencing_id"])["version_merged"].idxmax()) # indexes with latest version in seq_table
+idx.append(merged_table.loc[idx[0]].groupby(["profile_id"])["version_merged"].idxmax()) # indexes with latest version in seq_table
+
+all_rankings = list(configdata["priority_rankings"]["rankings"].keys())
```

Comment:
Here (and in the loop below) are we relying on the ordering of keys in the json? I don't think python dictionary safely preserves order, so we might want to add something more explicit to make sure the ordering is what we want